library/core/src/cell/lazy.rs

```diff
@@ -56,6 +56,31 @@ impl<T, F: FnOnce() -> T> LazyCell<T, F> {
         LazyCell { cell: OnceCell::new(), init: Cell::new(Some(init)) }
     }
 
+    /// Consumes the lazy, returning the evaluated value.
+    ///
+    /// Returns `Err(init)` if the lazy was never evaluated.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(once_cell)]
+    ///
+    /// use std::cell::LazyCell;
+    ///
+    /// let hello = "Hello, World!".to_string();
+    ///
+    /// let lazy = LazyCell::new(|| hello.to_uppercase());
+    ///
+    /// assert_eq!(&*lazy, "HELLO, WORLD!");
+    /// assert_eq!(LazyCell::into_inner(lazy).ok(), Some("HELLO, WORLD!".to_string()));
+    /// ```
+    #[unstable(feature = "once_cell", issue = "74465")]
+    pub fn into_inner(this: Self) -> Result<T, F> {
+        this.cell.into_inner().ok_or_else(|| {
+            this.init.take().unwrap_or_else(|| panic!("Lazy instance has previously been poisoned"))
```

Comment:
The panic should probably be mentioned in the docs

Reply:
None of the other methods mention panics, so I'd rather be consistent.